Index: src/main/java/com/colourtable/ColourTable.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.colourtable;\r\n\r\n\r\nimport java.util.ArrayList;\r\nimport java.util.List;\r\n\r\npublic class ColourTable {\r\n    private final int paletteSize;\r\n    private final List<Integer> palette;\r\n\r\n    public ColourTable(int paletteSize) {\r\n        if (!isValidPaletteSize(paletteSize)) {\r\n            throw new IllegalArgumentException(\"Invalid palette size. Must be between 2 and 1024\");\r\n        }\r\n        this.paletteSize = paletteSize;\r\n        this.palette = new ArrayList<>(paletteSize);\r\n    }\r\n\r\n    public int getPaletteSize() {\r\n        return paletteSize;\r\n    }\r\n\r\n    private boolean isValidPaletteSize(int size) {\r\n        return size > 1 && size < 1025;\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/colourtable/ColourTable.java b/src/main/java/com/colourtable/ColourTable.java
--- a/src/main/java/com/colourtable/ColourTable.java	(revision 79d430e122af6eed32abad79d53bc9b9ac2d73f5)
+++ b/src/main/java/com/colourtable/ColourTable.java	(date 1701209783323)
@@ -21,6 +21,6 @@
     }
 
     private boolean isValidPaletteSize(int size) {
-        return size > 1 && size < 1025;
+        return size > 1 && (size & (size - 1)) == 0 && size < 1025;
     }
 }
